### PR TITLE
214 toast component

### PR DESCRIPTION
**Pull request by WickyNilliams**
### What is the context of this PR?
Adds Toast and ToastList components. 

### How to review 
Review code, run tests, play with stories

### CODE REVIEW

File: src/components/Toast/Transition.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import styled from "styled-components";
+import { CSSTransition } from "react-transition-group";
+import PropTypes from "prop-types";
+
+const ToastTransition = styled(CSSTransition).attrs({
+  classNames: "toast"
+})`
+  transition: transform ${props => props.timeout}ms ease-out,
+    opacity ${props => props.timeout}ms ease-out;
```
Comment (hamishtaplin) [inline src/components/Toast/Transition.js]: Can we play around with this just to improve the sense of the toasts moving out of the way to accommodate new ones? Using delays/durations as discussed.

File: src/components/Toast/story.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+            <div>Hello</div>
+          </Toast>
+          <Toast id="2">
+            <div>World</div>
+          </Toast>
+        </ToastList>
+      </StoryInner>
+    </StoryContainer>
+  ))
+  .add("Interactive", () => <Story />);
```
Comment (hamishtaplin) [inline src/components/Toast/story.js]: Worth adding a "disappears on use" story to satisfy the relevant AC?

**Comment by samiwel** [comment]
Toast component looks good to me in general. Although I guess it wouldn't be a major issue in practice, wonder if we should limit the maximum number of toasts shown on the screen and queue up any extra ones. Otherwise you run the risk of having something like this. 
Too much toast?

**Comment by WickyNilliams** [comment]
@samiwel i can't imagine that happening in practice. In any case, I think logic for something like queuing is better placed outside of the Toast component itself, perhaps in a container component (e.g. `mapStateToProps` could select only the 5 oldest notifications at a time). This keeps Toast dumb but flexible

**Pull request merged**